### ISSUE TITLE
master: Ubuntu18 test failure: test-auxval

### ISSUE THREAD

**Issue by quanah** [heimdal/heimdal#446]
Master consistently fails for me on Ubuntu18 with test-auxval.  Error log has:

```
 cat test-suite.log
==============================================
   Heimdal 7.99.1: lib/roken/test-suite.log
==============================================

# TOTAL: 14
# PASS:  13
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: test-auxval
=================

test-auxval: rk_getauxval(16) should have been 126548989, was 2
auxv type 33 -> 140724198285312
FAIL test-auxval (exit status: 1)
```

**Comment by quanah** [comment]
Note: Fails both with and without optimization (-O2)

**Comment by quanah** [comment]
Note: Passes on both CentOS7 and Ubuntu16, so this issue seems very specific to Ubuntu18.  A compiler issue?

**Comment by rtandy** [comment]
Also seeing this on Debian unstable. Compiler here is ```gcc version 8.2.0 (Debian 8.2.0-13)```

**Comment by quanah** [comment]
For Ubuntu 18, compiler version is:

```
 gcc --version
gcc (Ubuntu 7.3.0-27ubuntu1~18.04) 7.3.0
Copyright (C) 2017 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by lhoward** [comment]
Probably more likely to be a kernel or libc issue than a compiler issue

**Comment by nicowilliams** [comment]
Aux vector entry `16` is `AT_HWCAP` -- that's kinda special I bet.  Indeed, from glibc:

```
unsigned long int
__getauxval (unsigned long int type)
{
#ifdef HAVE_AUX_VECTOR
  ElfW(auxv_t) *p;
#endif

  if (type == AT_HWCAP)
    return GLRO(dl_hwcap);
  else if (type == AT_HWCAP2)
    return GLRO(dl_hwcap2);

#ifdef HAVE_AUX_VECTOR
  for (p = GLRO(dl_auxv); p->a_type != AT_NULL; p++)
    if (p->a_type == type)
      return p->a_un.a_val;
#endif

  __set_errno (ENOENT);
  return 0;
}
```

So we need to skip `AT_HWCAP` and `AT_HWCAP2` in the test.

**Comment by lhoward** [comment]
Installing Ubuntu 18.10 now to investigate.

**Comment by lhoward** [comment]
```
diff --git a/lib/roken/test-auxval.c b/lib/roken/test-auxval.c
index f76c713bb..7e19ee530 100644
--- a/lib/roken/test-auxval.c
+++ b/lib/roken/test-auxval.c
@@ -166,7 +166,8 @@ main(int argc, char **argv, char **env)
             if (a[1] != 0)
                 warnx("AT_NULL with non-zero value %lu?!", a[1]);
             continue;
-        }
+        } else if (a[0] == AT_HWCAP || a[0] == AT_HWCAP2)
+           continue; /* these are handled specially by libc */
 
         errno = EACCES;
 
```

**Comment by lhoward** [comment]
Fixed in 2974a984
